index.js

```diff
@@ -52,6 +52,19 @@ function transact (fn, cb) {
   })
 }
 
+// Re-usable code adds the handlers nicely
+const addHandler = (existingHandler, newHandler) => {
+  if (Array.isArray(existingHandler)) {
+    existingHandler.push(newHandler)
+  } else if (typeof existingHandler === 'function') {
+    existingHandler = [existingHandler, newHandler]
+  } else {
+    existingHandler = [newHandler]
+  }
+
+  return existingHandler
+}
```

Comment:
this can be written in a nicer way without reassigning variables and returning early